### PR TITLE
Socket listen

### PR DESCRIPTION
**Pull request by allevo**
Add socket option.
Add `pre-commit` package as devDependency

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -72,6 +75,36 @@ test('should start the server at the given prefix', t => {
   })
 })
 
+test('should start fastify at given socket path', t => {
+  t.plan(2)
+
+  const sockFile = path.join(__dirname, '/test.sock')
```
Comment (delvedor) [inline test.js]: Probably `path.resolve` is better :)
Checkout how we used it [here](https://github.com/fastify/point-of-view/blob/master/index.js) :)

Comment (allevo) [reply test.js]: done

File: cli.js [hunk 2]
```diff
@@ -77,6 +77,8 @@ function runFastify (opts) {
 
   if (opts.address) {
     fastify.listen(opts.port, opts.address, listen)
+  } if (opts.socket) {
```
Comment (delvedor) [inline cli.js]: `else if`?

**Comment by allevo** [comment]
Should we throw an error if the user specify `address`  or `port` with `socket`?

File: cli.js (hunk 2 above)
Comment (allevo) [reply cli.js]: Wow! strange... `standard` says nothing about it!

**Comment by delvedor** [comment]
> Should we throw an error if the user specify address  or port with socket?

Mh, I'm not sure, probably is enough to document the order of our checks.
`address > socket > port`.

**Comment by allevo** [comment]
I don't know why travis fails. Restarting the travis job fixes the problem

**Pull request merged**